src/types/fetch_retry.ts

```diff
@@ -0,0 +1,27 @@
+// This is copied from fetch-retry/index.d.ts
+// fetch-retry does not export these types
+
+export type RequestDelayFunction = (
+  attempt: number,
+  error: Error | null,
+  response: Response | null
+) => number;
+
+export type RequestRetryOnFunction = (
+  attempt: number,
+  error: Error | null,
+  response: Response | null
+) => boolean | Promise<boolean>;
+
+export interface RequestInitWithRetry extends RequestInit {
+  retries?: number;
+  retryDelay?: number | RequestDelayFunction;
+  retryOn?: number[] | RequestRetryOnFunction;
+}
+
+// This interface is created by us
+export interface RetryOptions {
+  retries?: number;
+  retryDelay?: number | RequestDelayFunction;
+  retryOn?: number[] | RequestRetryOnFunction;
+}
```

Comment:
Is there no `@types/fetch-retry` package? In any case, this should not be in the `src` folder. If there is no other option than copying this to our repository, we should put this in a `vendor` folder.

Reply:
fetch-retry includes these types, but does not export these, see https://github.com/jonbern/fetch-retry/pull/50
Agreed on moving these to a separate folder (note that `RetryOptions` is something I added and not something from fetch-retry, so I've kept this in the same location)

Reply:
Hmm, I would also move the `RetryOptions`. It's not a type of ours, even if we wrote it ourselves.

Reply:
It is something we devised to allow a read request to have these options. Since fetch-retry doesn't support this request with just this params (but requires them to be included in a request object), and the users of this library will have to interact with it, I would say that it is "our" type.

Reply:
Though in the end, it doesn't matter that much. But will show up in the generated `index.d.ts` just with a different location

Reply:
Ah, then I misunderstood what this type was about.